### PR TITLE
Add ability to save file while upload with original name instead of hash

### PR DESCRIPTION
**Pull request by mariuszsokal-bitbag**
| Q               | A
| --------------- | -----
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | none
| License         | MIT

### CODE REVIEW

File: src/Entity/Media.php [hunk 1]
```diff
@@ -58,6 +58,9 @@ class Media implements MediaInterface
     /** @var int|null */
     protected $height;
 
+    /** @var bool|null */
```
Comment (JakubWegner) [inline src/Entity/Media.php:61]: Maybe default false will be better instead of null?

File: src/Uploader/MediaUploader.php [hunk 2]
```diff
@@ -36,9 +36,17 @@ public function upload(MediaInterface $media, string $pathPrefix): void
             $this->remove($media->getPath());
         }
 
+        $originalName = null;
+        if ($media->getSaveWithOriginalName()) {
+            try {
+                $originalName = $file->getClientOriginalName();
```
Comment (JakubWegner) [inline src/Uploader/MediaUploader.php:42]: Looks like it's `File`, not `UploadedFile`, so maybe try `SplFileInfo::getFilename()` to get the original file name

File: src/Uploader/MediaUploader.php [hunk 3]
```diff
@@ -71,14 +79,18 @@ public function remove(string $path): bool
         return false;
     }
 
-    private function expandPath(string $path, string $pathPrefix): string
+    private function expandPath(
+        string $path,
+        string $pathPrefix,
+        ?string $originalName = null
+    ): string
```
Comment (JakubWegner) [inline src/Uploader/MediaUploader.php:86]: Please use the PSR style `): string {`

**Review by JakubWegner — CHANGES_REQUESTED** [review]

File: src/Entity/Media.php (hunk 1 above)
Comment (marekrzytki) [reply src/Entity/Media.php]: Sure, changed to default false.

File: src/Uploader/MediaUploader.php (hunk 3 above)
Comment (marekrzytki) [reply src/Uploader/MediaUploader.php]: Sure, fixed

File: src/Uploader/MediaUploader.php (hunk 2 above)
Comment (JakubWegner) [reply src/Uploader/MediaUploader.php:42]: Looks good for me!

**Review by JakubWegner — APPROVED** [review]

**Review by PiotrSzymanski2000 — APPROVED** [review]

**Review by liszkapawel — APPROVED** [review]

**Pull request merged**